{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "### Problem\n\nDescribe the bug\r\nI am not able to get neovim to detect python 3 on Windows 10 pro. I made several attempts.\r\n\r\ninstalled python 3.12 from the official site (before tried several times with 3.10, 3.11. 3.12 from the official site, winget and chocolatey)\r\ninstalled neovim 0.10 via winget \r\nfollowed the guide https://neovim.io/doc/user/provider.html . I am sure I used correct path, keys, etc.\r\nnvim has never detected a python provider able to import neovim (tested within several shells)\r\n\r\nEvery time I ran :checkhealth and got this:\r\n```\r\nprovider.python: require(\"provider.python.health\").check()\r\n\r\nPython 3 provider (optional) ~\r\n- `g:python3_host_prog` is not set. Searching for python in the environment.\r\n- Executable: C:\\Users\\ASUS\\AppData\\Local\\Programs\\Python\\Python312\\python.EXE\r\n- Python version: 3.12.4\r\n- pynvim version: 0.5.0\r\n- WARNING Could not contact PyPI to get latest version.\r\n- ERROR HTTP request failed: error: curl error with https://pypi.python.org/pypi/pynvim/json: 35\r\n\r\nPython virtualenv ~\r\n- OK no $VIRTUAL_ENV\r\n```\r\nNOTE: the Json problem gives 404 error \r\nLater I tried to config with the path but nvim can't detect the executable : \r\n```\r\nprovider.python: require(\"provider.python.health\").check()\r\n\r\nPython 3 provider (optional) ~\r\n- Using: g:python3_host_prog = \"C:\\Users\\ASUS\\AppData\\Local\\Programs\\Python\\Python312\\\"\r\n- Executable: Not found\r\n- ERROR Detected pip upgrade failure: Python executable can import \"pynvim\" but not \"neovim\": \r\n  - ADVICE:\r\n    - Use that Python version to reinstall \"pynvim\" and optionally \"neovim\".\r\n    -  -m pip uninstall pynvim neovim\r\n    -  -m pip install pynvim\r\n    -  -m pip install neovim  # only if needed by third-party software\r\n``` \n\n### Steps to reproduce\n\n- install neovim with winget\r\n- install python with different way ( winget, chocolatey, official site intaller)\r\n- use CMD,  powershell\r\n- checkhealth\r\n\n\n### Expected behavior\n\nI would expect neovim to detect the provider \n\n### Neovim version (nvim -v)\n\n0.10\n\n### Vim (not Nvim) behaves the same?\n\nnot installed\n\n### Operating system/version\n\nWindows 10\n\n### Terminal name/version\n\nWindows Terminal, CMD, powershell\n\n### $TERM environment variable\n\nNone\n\n### Installation\n\nWinget, Chocolatey, Installler",
    "closed_at": "2024-06-18T23:03:53Z",
    "comment_data": [
        {
            "author_association": "MEMBER",
            "body": "> ERROR HTTP request failed: error: curl error with https://pypi.python.org/pypi/pynvim/json: 35\r\n\r\nWhy is that HTTP request failing? https://github.com/neovim/neovim/blob/20a7eebec086129e605041d32916f36df50890de/runtime/lua/vim/provider/health.lua#L439\r\n\r\nDoes `curl https://pypi.python.org/pypi/pynvim/json` work for you?\r\n\r\n> Using: g:python3_host_prog = \"C:\\Users\\ASUS\\AppData\\Local\\Programs\\Python\\Python312\\\"\r\n\r\n`g:python3_host_prog` must point to the exe, not the directory.",
            "created_at": "2024-06-17T22:49:43Z",
            "html_url": "https://github.com/neovim/neovim/issues/29336#issuecomment-2174568420",
            "id": 2174568420,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/29336",
            "node_id": "IC_kwDOAPphoM6BnUfk",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2174568420/reactions"
            },
            "updated_at": "2024-06-17T22:49:43Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2174568420",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1359421?v=4",
                "events_url": "https://api.github.com/users/justinmk/events{/privacy}",
                "followers_url": "https://api.github.com/users/justinmk/followers",
                "following_url": "https://api.github.com/users/justinmk/following{/other_user}",
                "gists_url": "https://api.github.com/users/justinmk/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/justinmk",
                "id": 1359421,
                "login": "justinmk",
                "node_id": "MDQ6VXNlcjEzNTk0MjE=",
                "organizations_url": "https://api.github.com/users/justinmk/orgs",
                "received_events_url": "https://api.github.com/users/justinmk/received_events",
                "repos_url": "https://api.github.com/users/justinmk/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/justinmk/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/justinmk/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/justinmk"
            }
        },
        {
            "author_association": "NONE",
            "body": "> ok so now I set ( let g:python3_host_prog = 'C:\\Users\\ASUS\\AppData\\Local\\Programs\\Python\\Python312\\python.exe and python.EXE' in both cases appeared this:\r\n```\r\n- g:python3_host_prog is not set. Searching for python in the environment.\r\n- Executable: C:\\Users\\ASUS\\AppData\\Local\\Programs\\Python\\Python312\\python.EXE\r\n``` \r\nas for   ```curl https://pypi.python.org/pypi/pynvim/json``` it gives me:\r\n \r\n```\r\n curl: (35) schannel: next InitializeSecurityContext failed: CRYPT_E_NO_REVOCATION_CHECK (0x80092012) - The revocation function was unable to check revocation for the certificate.\r\n``` \r\n",
            "created_at": "2024-06-18T05:02:12Z",
            "html_url": "https://github.com/neovim/neovim/issues/29336#issuecomment-2175007589",
            "id": 2175007589,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/29336",
            "node_id": "IC_kwDOAPphoM6Bo_tl",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2175007589/reactions"
            },
            "updated_at": "2024-06-18T06:37:28Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2175007589",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/113310442?v=4",
                "events_url": "https://api.github.com/users/JDCerberus/events{/privacy}",
                "followers_url": "https://api.github.com/users/JDCerberus/followers",
                "following_url": "https://api.github.com/users/JDCerberus/following{/other_user}",
                "gists_url": "https://api.github.com/users/JDCerberus/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/JDCerberus",
                "id": 113310442,
                "login": "JDCerberus",
                "node_id": "U_kgDOBsD66g",
                "organizations_url": "https://api.github.com/users/JDCerberus/orgs",
                "received_events_url": "https://api.github.com/users/JDCerberus/received_events",
                "repos_url": "https://api.github.com/users/JDCerberus/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/JDCerberus/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/JDCerberus/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/JDCerberus"
            }
        },
        {
            "author_association": "NONE",
            "body": "> > ok so now I set ( let g:python3_host_prog = 'C:\\Users\\ASUS\\AppData\\Local\\Programs\\Python\\Python312\\python.exe and python.EXE' in both cases appeared this:\r\n> \r\n> ```\r\n> - g:python3_host_prog is not set. Searching for python in the environment.\r\n> - Executable: C:\\Users\\ASUS\\AppData\\Local\\Programs\\Python\\Python312\\python.EXE\r\n> ```\r\n> \r\n> as for `curl https://pypi.python.org/pypi/pynvim/json` it gives me:\r\n> \r\n> ```\r\n>  curl: (35) schannel: next InitializeSecurityContext failed: CRYPT_E_NO_REVOCATION_CHECK (0x80092012) - The revocation function was unable to check revocation for the certificate.\r\n> ```\r\n\r\nSo this is a little after, I looked to understand the problem and I found that the connection problem may be due problems with  ```network restrictions or firewalls``` , then I thought **_Antivirus_** I disabled it,  and no more HTTP problems .\r\n\r\nBut the first problem still exist and other problems appeared: \r\n```\r\n Python 3 provider (optional) ~\r\n- Using: g:python3_host_prog = \"C:UsersASUSAppDataLocalProgramsPythonPython312python.exe\"\r\n- Executable: Not found\r\n- ERROR Detected pip upgrade failure: Python executable can import \"pynvim\" but not \"neovim\": \r\n  - ADVICE:\r\n    - Use that Python version to reinstall \"pynvim\" and optionally \"neovim\".\r\n    -  -m pip uninstall pynvim neovim\r\n    -  -m pip install pynvim\r\n    -  -m pip install neovim  # only if needed by third-party software\r\n```\r\n> here I'm using vim.g.python3_host_prog and formatted the direction without ```\\``` \r\nI have ``` pynvim and neovim``` installed with pip",
            "created_at": "2024-06-18T05:30:37Z",
            "html_url": "https://github.com/neovim/neovim/issues/29336#issuecomment-2175044851",
            "id": 2175044851,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/29336",
            "node_id": "IC_kwDOAPphoM6BpIzz",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2175044851/reactions"
            },
            "updated_at": "2024-06-18T19:46:23Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2175044851",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/113310442?v=4",
                "events_url": "https://api.github.com/users/JDCerberus/events{/privacy}",
                "followers_url": "https://api.github.com/users/JDCerberus/followers",
                "following_url": "https://api.github.com/users/JDCerberus/following{/other_user}",
                "gists_url": "https://api.github.com/users/JDCerberus/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/JDCerberus",
                "id": 113310442,
                "login": "JDCerberus",
                "node_id": "U_kgDOBsD66g",
                "organizations_url": "https://api.github.com/users/JDCerberus/orgs",
                "received_events_url": "https://api.github.com/users/JDCerberus/received_events",
                "repos_url": "https://api.github.com/users/JDCerberus/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/JDCerberus/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/JDCerberus/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/JDCerberus"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "> here I'm using vim.g.python3_host_prog and formatted the direction without `\\`\r\n\r\nWhat does this mean? Show actual code from your config, don't describe things. Just show the code.\r\n\r\nI'm guessing that your config has something like this:\r\n\r\n    vim.g.python3_host_prog = \"C:\\Users\\ASUS\\AppData\\Local\\Programs\\Python\\Python312\\\"\r\n\r\nThat is why checkhealth reports that `\"C:UsersASUSAppDataLocalProgramsPythonPython312python.exe\"` cannot be found.\r\n\r\n1. on Windows you don't need to use `\\` slashes. `/` slashes work just fine, except for some cmd.exe commands.\r\n   ```\r\n   vim.g.python3_host_prog = \"C:/Users/ASUS/AppData/Local/Programs/Python/Python312/\"\r\n   ```\r\n2. if you really want to use `\\` slashes you need to escape them:\r\n   ```\r\n   vim.g.python3_host_prog = \"C:\\\\Users\\\\ASUS\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python312\\\\\"\r\n   ```",
            "created_at": "2024-06-18T23:03:53Z",
            "html_url": "https://github.com/neovim/neovim/issues/29336#issuecomment-2177226422",
            "id": 2177226422,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/29336",
            "node_id": "IC_kwDOAPphoM6Bxda2",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2177226422/reactions"
            },
            "updated_at": "2024-06-18T23:07:08Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2177226422",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1359421?v=4",
                "events_url": "https://api.github.com/users/justinmk/events{/privacy}",
                "followers_url": "https://api.github.com/users/justinmk/followers",
                "following_url": "https://api.github.com/users/justinmk/following{/other_user}",
                "gists_url": "https://api.github.com/users/justinmk/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/justinmk",
                "id": 1359421,
                "login": "justinmk",
                "node_id": "MDQ6VXNlcjEzNTk0MjE=",
                "organizations_url": "https://api.github.com/users/justinmk/orgs",
                "received_events_url": "https://api.github.com/users/justinmk/received_events",
                "repos_url": "https://api.github.com/users/justinmk/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/justinmk/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/justinmk/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/justinmk"
            }
        },
        {
            "author_association": "NONE",
            "body": "Hi, that resolved the problem, I'm sorry for the inconvenience, this is the first time I use GitHub and lot of  similar problems  I found had ```\\``` so I didn't know.\r\nSorry again if I made you lost time and thank you for the help",
            "created_at": "2024-06-19T02:22:59Z",
            "html_url": "https://github.com/neovim/neovim/issues/29336#issuecomment-2177415973",
            "id": 2177415973,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/29336",
            "node_id": "IC_kwDOAPphoM6ByLsl",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2177415973/reactions"
            },
            "updated_at": "2024-06-19T02:24:02Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2177415973",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/113310442?v=4",
                "events_url": "https://api.github.com/users/JDCerberus/events{/privacy}",
                "followers_url": "https://api.github.com/users/JDCerberus/followers",
                "following_url": "https://api.github.com/users/JDCerberus/following{/other_user}",
                "gists_url": "https://api.github.com/users/JDCerberus/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/JDCerberus",
                "id": 113310442,
                "login": "JDCerberus",
                "node_id": "U_kgDOBsD66g",
                "organizations_url": "https://api.github.com/users/JDCerberus/orgs",
                "received_events_url": "https://api.github.com/users/JDCerberus/received_events",
                "repos_url": "https://api.github.com/users/JDCerberus/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/JDCerberus/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/JDCerberus/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/JDCerberus"
            }
        }
    ],
    "comments": 5,
    "comments_url": "https://api.github.com/repos/neovim/neovim/issues/29336/comments",
    "created_at": "2024-06-14T17:01:35Z",
    "events_url": "https://api.github.com/repos/neovim/neovim/issues/29336/events",
    "html_url": "https://github.com/neovim/neovim/issues/29336",
    "id": 2353763386,
    "labels": [
        {
            "color": "e6e6e6",
            "default": false,
            "description": "issues that are closed as usage questions",
            "id": 77997478,
            "name": "closed:question",
            "node_id": "MDU6TGFiZWw3Nzk5NzQ3OA==",
            "url": "https://api.github.com/repos/neovim/neovim/labels/closed:question"
        },
        {
            "color": "d4c5f9",
            "default": false,
            "description": null,
            "id": 109461219,
            "name": "platform:windows",
            "node_id": "MDU6TGFiZWwxMDk0NjEyMTk=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/platform:windows"
        },
        {
            "color": "c5def5",
            "default": false,
            "description": null,
            "id": 144672485,
            "name": "provider",
            "node_id": "MDU6TGFiZWwxNDQ2NzI0ODU=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/provider"
        }
    ],
    "labels_url": "https://api.github.com/repos/neovim/neovim/issues/29336/labels{/name}",
    "locked": false,
    "milestone": null,
    "node_id": "I_kwDOAPphoM6MS5Q6",
    "number": 29336,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/neovim/neovim/issues/29336/reactions"
    },
    "repository_url": "https://api.github.com/repos/neovim/neovim",
    "state": "closed",
    "state_reason": "not_planned",
    "timeline_url": "https://api.github.com/repos/neovim/neovim/issues/29336/timeline",
    "title": "can't detect python provider in Windows 10",
    "updated_at": "2024-06-19T02:24:02Z",
    "url": "https://api.github.com/repos/neovim/neovim/issues/29336",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/113310442?v=4",
        "events_url": "https://api.github.com/users/JDCerberus/events{/privacy}",
        "followers_url": "https://api.github.com/users/JDCerberus/followers",
        "following_url": "https://api.github.com/users/JDCerberus/following{/other_user}",
        "gists_url": "https://api.github.com/users/JDCerberus/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/JDCerberus",
        "id": 113310442,
        "login": "JDCerberus",
        "node_id": "U_kgDOBsD66g",
        "organizations_url": "https://api.github.com/users/JDCerberus/orgs",
        "received_events_url": "https://api.github.com/users/JDCerberus/received_events",
        "repos_url": "https://api.github.com/users/JDCerberus/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/JDCerberus/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/JDCerberus/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/JDCerberus"
    }
}